api/src/main/java/org/openmrs/api/db/hibernate/HibernateCohortDAO.java

```diff
@@ -51,26 +53,42 @@ public void setSessionFactory(SessionFactory sessionFactory) {
 	public Cohort getCohort(Integer id) throws DAOException {
 		return (Cohort) sessionFactory.getCurrentSession().get(Cohort.class, id);
 	}
-	
+
+	/**
+	 * @see org.openmrs.api.db.CohortDAO#getCohortsContainingPatientId(java.lang.Integer)
+	 */
+	@SuppressWarnings("unchecked")
+	public List<Cohort> getCohortsContainingPatientId(Integer patientId, Boolean voided) throws DAOException {
+		Disjunction orDisjunction = Restrictions.disjunction();
+		orDisjunction.add(Restrictions.isNull("m.endDate"));
+		orDisjunction.add(Restrictions.gt("m.endDate", new Date()));
+		return (List<Cohort>) sessionFactory.getCurrentSession().createCriteria(Cohort.class)
+				.add(Restrictions.eq("voided", voided))
+				.createAlias("members", "m")
+				.add(Restrictions.le("m.startDate", new Date()))
+				.add(orDisjunction)
+				.createAlias("m.patient", "p")
+				.add(Restrictions.eq("p.patientId", patientId))
```

Comment:
So I found out that I could query patient #6 but not patient #7. The only difference I see here is Cohort property is null for patient #7. Patient #7 is the new added member. I could fix this by adding Cohort to the CohortMembership constructor (the way it was working earlier). Is there any other workaround to fix this?

<img width="478" alt="screen shot 2016-11-06 at 12 53 23 pm" src="https://cloud.githubusercontent.com/assets/193819/20041572/2334c82e-a420-11e6-9dd2-cb6a2085538e.png">